src/mavsdk_server/src/grpc_server.cpp

```diff
@@ -19,33 +21,126 @@ int GrpcServer::run()
     setup_port(builder);
 
     builder.RegisterService(&_core);
+
+#ifdef ACTION_ENABLED
     builder.RegisterService(&_action_service);
+#endif
+
+#ifdef ACTION_SERVER_ENABLED
     builder.RegisterService(&_action_server_service);
+#endif
+
+#ifdef CALIBRATION_ENABLED
     builder.RegisterService(&_calibration_service);
+#endif
+
+#ifdef CAMERA_ENABLED
     builder.RegisterService(&_camera_service);
+#endif
+
+#ifdef CAMERA_SERVER_ENABLED
```

Comment:
It looks a bit not elegant, maybe the 3 lines can be wrapped into some macro Call like REGISTER_IF_ENABLE(…) but it’s auto generated so also does not matter so much.

Reply:
Right, it's not pretty but it's just auto-generated, and at least it's not out of sync like it often was in the past because someone would add a plugin and they and us maintainers would both forget to update it.

The plugin header includes are not there at all, so not sure how it would be there.

Reply:
Yeah I agree.  The only way i could imagine moving this into the plugin would be through lazy plugin somehow, am not sure if that’s possible, is also rather cosmetic and not so important.

Reply:
I don't quite understand how you imagine it. Feel free to make a PR or code example to show it. This can also happen later, of course.

Reply:
Something like this https://diffy.org/diff/43d4582733066

Reply:
@devbharat I think what you suggest makes sense, thanks!

@JonasVautherin can you check that change in diffy as well? You know this part of the code.

Reply:
I'll check ASAP :+1:

Reply:
I was also not sure of that, but the unit test thing seems to work. Worst case we can have like a two step constructor that creates the service and registers it and then returns the object.